File: android_mobile/src/main/java/com/alexstyl/specialdates/theming/ThemingModule.java
package com.alexstyl.specialdates.theming;

import android.content.Context;
import android.content.res.Resources;

import dagger.Module;
import dagger.Provides;

@Module
public class ThemingModule {

    @Provides
    Themer themer(ThemingPreferences preferences, AttributeExtractor attributeExtractor, Resources resources) {
        return new Themer(preferences, attributeExtractor, resources);
    }

    @Provides
    AttributeExtractor attributeExtractor() {
        return new AttributeExtractor();
    }

    @Provides
    ThemingPreferences themingPreferences(Context context) {
        return ThemingPreferences.Companion.newInstance(context);
    }

}

File: android_mobile/src/main/java/com/alexstyl/specialdates/theming/ThemingModule.kt
package com.alexstyl.specialdates.theming

import android.content.Context
import android.content.res.Resources

import com.alexstyl.specialdates.EasyPreferences

import dagger.Module
import dagger.Provides

@Module
class ThemingModule {

    @Provides
    internal fun themer(preferences: ThemingPreferences, attributeExtractor: AttributeExtractor, resources: Resources): Themer {
        return Themer(preferences, attributeExtractor, resources)
    }

    @Provides
    internal fun attributeExtractor(): AttributeExtractor {
        return AttributeExtractor()
    }

    @Provides
    internal fun themingPreferences(context: Context): ThemingPreferences {
        val defaultPreferences = EasyPreferences.createForDefaultPreferences(context)
        return ThemingPreferences(defaultPreferences)
    }

}
